### PR TITLE
band integrated

### PR DESCRIPTION
**Pull request by cgsingh33**

### CODE REVIEW

File: proto/comdex/rewards/v1beta1/query.proto [hunk 1]
```diff
@@ -173,6 +176,19 @@ message QueryWhitelistedAppIdsVaultResponse {
...
+}
+
+message QueryExternalRewardLendsRequest{
+  cosmos.base.query.v1beta1.PageRequest pagination = 1
+  [(gogoproto.moretags) = "yaml:\"pagination\""];
+}
+
+message QueryExternalRewardLendsResponse {
+  repeated Lend_external_rewards lend_external_rewards = 1
+  [(gogoproto.nullable) = false, (gogoproto.moretags) = "yaml:\"vault_external_rewards\""];
```
Comment (dheerajkd30) [inline proto/comdex/rewards/v1beta1/query.proto]: "yaml:\"vault_external_rewards this should be changed

File: proto/comdex/market/v1beta1/query.proto [hunk 2]
```diff
@@ -17,22 +17,22 @@ message QueryMarketsRequest {
 }
 
 message QueryMarketsResponse {
-  repeated Market markets = 1 [
+  repeated TimeWeightedAverage timeWeightedAverage = 1 [
     (gogoproto.nullable) = false,
-    (gogoproto.moretags) = "yaml:\"markets\""
+    (gogoproto.moretags) = "yaml:\"timeWeightedAverage\""
```
Comment (dheerajkd30) [inline proto/comdex/market/v1beta1/query.proto]: change to snake case

File: proto/comdex/market/v1beta1/query.proto [hunk 3]
```diff
@@ -17,22 +17,22 @@ message QueryMarketsRequest {
...
-  string symbol = 1 [(gogoproto.moretags) = "yaml:\"symbol\""];
+  uint64 assetID = 1 [(gogoproto.moretags) = "yaml:\"assetID\""];
 }
 
 message QueryMarketResponse {
-  Market market = 1 [
+  TimeWeightedAverage timeWeightedAverage = 1 [
     (gogoproto.nullable) = false,
-    (gogoproto.moretags) = "yaml:\"market\""
+    (gogoproto.moretags) = "yaml:\"timeWeightedAverage\""
```
Comment (dheerajkd30) [inline proto/comdex/market/v1beta1/query.proto]: timeWeightedAverage in yaml  to snake case

File: proto/comdex/market/v1beta1/query.proto (hunk 2 above)
Comment (cgsingh33) [reply proto/comdex/market/v1beta1/query.proto]: done

File: proto/comdex/rewards/v1beta1/query.proto (hunk 1 above)
Comment (cgsingh33) [reply proto/comdex/rewards/v1beta1/query.proto]: done

File: proto/comdex/market/v1beta1/query.proto (hunk 3 above)
Comment (cgsingh33) [reply proto/comdex/market/v1beta1/query.proto]: done

File: proto/comdex/market/v1beta1/query.proto [hunk 4]
```diff
@@ -17,22 +17,22 @@ message QueryMarketsRequest {
...
-    (gogoproto.moretags) = "yaml:\"markets\""
+    (gogoproto.moretags) = "yaml:\"time_weighted_average\""
   ];
   cosmos.base.query.v1beta1.PageResponse pagination = 2
   [(gogoproto.moretags) = "yaml:\"pagination\""];
 }
 
 message QueryMarketRequest {
-  string symbol = 1 [(gogoproto.moretags) = "yaml:\"symbol\""];
+  uint64 assetID = 1 [(gogoproto.moretags) = "yaml:\"assetID\""];
```
Comment (dheerajkd30) [inline proto/comdex/market/v1beta1/query.proto]: again asset_id

Comment (cgsingh33) [reply proto/comdex/market/v1beta1/query.proto]: done

File: x/esm/keeper/keeper.go [hunk 5]
```diff
@@ -182,34 +188,35 @@ func (k Keeper) CalculateCollateral(ctx sdk.Context, appID uint64, amount sdk.Co
...
+	// initializing userWorth
 	userWorth := sdk.ZeroDec()
 	for _, v := range esmDataAfterCoolOff.DebtAsset {
 		if v.AssetID == assetInID.Id {
 			userWorth = v.DebtTokenWorth.Mul(sdk.NewDecFromInt(amount.Amount))
 			break
 		}
 	}
-
+	// TODO: Refactor & remove usd
```
Comment (dheerajkd30) [inline x/esm/keeper/keeper.go]: when will this be removed ? in future releases ?

**Comment by vishnukumavat** [comment]
@cgsingh33 @rajrohit10 
```
func (k Keeper) CalculateTwa(ctx sdk.Context, twa types.TimeWeightedAverage) uint64 {
	var sum uint64
	for _, price := range twa.PriceValue {
		sum += price
	}
	twa.Twa = sum / 30
	return twa.Twa
}
```

can you please explain why it is divided by 30 instead of ```len(twa.PriceValue)``` ?

File: proto/comdex/market/v1beta1/genesis.proto [hunk 6]
```diff
@@ -9,8 +9,8 @@ option (gogoproto.equal_all)           = false;
 option (gogoproto.goproto_getters_all) = false;
 
 message GenesisState {
-    repeated Market markets = 1 [
-        (gogoproto.moretags) = "yaml:\"markets\"",
+    repeated TimeWeightedAverage timeWeightedAverage = 1 [
```
Comment (vishnukumavat) [inline proto/comdex/market/v1beta1/genesis.proto]: snake case please - time_weighted_average

File: proto/comdex/market/v1beta1/genesis.proto [hunk 7]
```diff
@@ -9,8 +9,8 @@ option (gogoproto.equal_all)           = false;
...
+        (gogoproto.moretags) = "yaml:\"timeWeightedAverage\"",
```
Comment (vishnukumavat) [inline proto/comdex/market/v1beta1/genesis.proto]: this as well

File: proto/comdex/rewards/v1beta1/rewards.proto [hunk 8]
```diff
@@ -162,16 +162,16 @@ message Lend_external_rewards{
   uint64 app_mapping_id = 2 [
     (gogoproto.moretags) = "yaml:\"app_mapping_id\""
   ];
-  repeated Rewards_asset_pool_data rewards_asset_pool_data = 3 [
+  Rewards_asset_pool_data rewards_asset_pool_data = 3 [
```
Comment (vishnukumavat) [inline proto/comdex/rewards/v1beta1/rewards.proto]: Msg/struct names in CamelCase please, don't use snake case here

Comment (vishnukumavat) [reply proto/comdex/rewards/v1beta1/rewards.proto]: also, please refactor in entire file wherever naming is done this way

File: proto/comdex/rewards/v1beta1/tx.proto [hunk 9]
```diff
@@ -131,7 +132,7 @@ message ActivateExternalRewardsVault {
     (gogoproto.moretags) = "yaml:\"app_mapping_id\""
   ];
   uint64 extended_Pair_Id = 2 [
```
Comment (vishnukumavat) [inline proto/comdex/rewards/v1beta1/tx.proto]: extended_Pair_Id should be extended_pair_id, please update

File: proto/comdex/rewards/v1beta1/tx.proto [hunk 10]
```diff
@@ -149,4 +150,41 @@ message ActivateExternalRewardsVault {
...
 
-message ActivateExternalRewardsVaultResponse {}
\ No newline at end of file
+message ActivateExternalRewardsVaultResponse {}
+
+message ActivateExternalRewardsLend {
+  uint64 app_mapping_id = 1 [
+    (gogoproto.moretags) = "yaml:\"app_mapping_id\""
+  ];
+  uint64 c_pool_Id = 2 [
```
Comment (vishnukumavat) [inline proto/comdex/rewards/v1beta1/tx.proto]: should be c_pool_id

File: x/esm/keeper/keeper.go (hunk 5 above)
Comment (pratikasr) [reply x/esm/keeper/keeper.go]: TODO removed

File: x/vault/client/testutil/helpers.go [hunk 11]
```diff
@@ -122,20 +108,42 @@ func (s *VaultIntegrationTestSuite) CreateNewAsset(name, denom string, price uin
 	}
 	s.Require().NotZero(assetID)
 
-	market := markettypes.Market{
-		Symbol:   name,
-		ScriptID: 12,
-		Rates:    price,
+	twa1 := markettypes.TimeWeightedAverage{
```
Comment (vishnukumavat) [inline x/vault/client/testutil/helpers.go:111]: shouldn't be this way, please refer fix from the liquidity module. 
this causes test case failure

File: x/vault/keeper/keeper_test.go [hunk 12]
```diff
@@ -132,20 +118,42 @@ func (s *KeeperTestSuite) CreateNewAsset(name, denom string, price uint64) uint6
 	}
 	s.Require().NotZero(assetID)
 
-	market := markettypes.Market{
-		Symbol:   name,
-		ScriptID: 12,
-		Rates:    price,
+	twa1 := markettypes.TimeWeightedAverage{
```
Comment (vishnukumavat) [inline x/vault/keeper/keeper_test.go:121]: please change this as well, for reference look liquidity module's change in the same PR.

File: x/vault/module_test.go [hunk 13]
```diff
@@ -118,20 +104,42 @@ func (s *ModuleTestSuite) CreateNewAsset(name, denom string, price uint64) uint6
 	}
 	s.Require().NotZero(assetID)
 
-	market := markettypes.Market{
-		Symbol:   name,
-		ScriptID: 12,
-		Rates:    price,
+	twa1 := markettypes.TimeWeightedAverage{
```
Comment (vishnukumavat) [inline x/vault/module_test.go:107]: please use the liquidity module's test case fix for reference for this.

**Review by dheerajkd30 — APPROVED** [review]

**Pull request merged**